src/rust/ide/json-rpc/src/macros.rs

```diff
@@ -82,22 +82,29 @@ macro_rules! make_rpc_methods {
         }
 
         impl API for Client {
-            $(fn $method(&self, $($param_name:$param_ty),*)
+            $(fn $method<'a>(&'a self, $($param_name:&'a $param_ty),*)
             -> std::pin::Pin<Box<dyn Future<Output=Result<$result>>>> {
-                let input = $method_input { $($param_name:$param_name),* };
-                Box::pin(self.handler.borrow().open_request(input))
+                use json_rpc::api::RemoteMethodCall;
+                let phantom    = std::marker::PhantomData;
+                let input      = $method_input { phantom, $($param_name:&$param_name),* };
+                let input_json = serde_json::to_value(input).unwrap();
+                let name       = $method_input::NAME;
+                let result_fut = self.handler.borrow().open_request_with_json(name,&input_json);
+                Box::pin(result_fut)
             })*
         }
 
         $(
             /// Structure transporting method arguments.
-            #[derive(Serialize,Deserialize,Debug,PartialEq)]
+            #[derive(Serialize,Debug,PartialEq)]
             #[serde(rename_all = "camelCase")]
-            struct $method_input {
-                $($param_name : $param_ty),*
+            struct $method_input<'a> {
+                #[serde(skip)]
+                phantom : std::marker::PhantomData<&'a u8>,
```

Comment:
When I saw this code I wondered "why u8?", perhaps unit gives a stronger suggestion that this is for lifetime alone :)
```suggestion
                phantom : std::marker::PhantomData<&'a()>,
```